### PR TITLE
Fix ub in unit tests; performance tests for xint; added libboost-thread-dev in install.sh

### PR DESCRIPTION
**Pull request by mikurys**

### CODE REVIEW

File: src/test/xint.cpp [hunk 1]
```diff
@@ -425,10 +425,15 @@ template <typename T_INT> bool is_safe_int() {
...
+	bool safetype = is_safe_int<T_INT>(); // is this given type T_INT a safe type, so we can try to overflow it,
+	// and doing such test will not be an UB
 	// @hint: if this fails, then it can be called from code "generate_tests_for_types" go fix that
 	T_INT a = val;	t_correct_int a_corr = a;
-	auto func = [&]() { a_corr+=1; a++; } ;
+	auto func = [&]() {
+		a_corr+=1;
+		a++;
+	} ;
 	#define db do { _mark("safe="<<safetype<<"; a="<<a<<" a_corr="<<a_corr); } while(0)
```
Comment (robertoleksy) [inline src/test/xint.cpp]: remove this macro

File: src/test/xint.cpp [hunk 2]
```diff
@@ -438,25 +443,39 @@ void math_tests_overflow_incr(T_INT val) { bool safetype = is_safe_int<T_INT>();
...
+	// and doing such test will not be an UB
+
 	T_INT a = val;	t_correct_int a_corr = a;
 	EXPECT_EQ(a,a_corr);
-	auto func = [&]() { a_corr-=1; a--; } ;
+	auto func = [&]() {
+		a_corr-=1;
+		a--;
+	} ;
 	#define db do { _mark("safe="<<safetype<<"; a="<<a<<" a_corr="<<a_corr); } while(0)
```
Comment (robertoleksy) [inline src/test/xint.cpp]: remove this macro

File: src/test/xint.cpp [hunk 3]
```diff
@@ -493,7 +512,7 @@ TEST(xint, math1) {
 
 //const test_xint::detail::t_correct_int maxni = 0xFFFFFFFFFFFFFFFF; // max "normal integer" on this platform
 #define maxni 0xFFFFFFFFFFFFFFFFULL // max "normal integer" on this platform
-
+#define maxni_half 0x7FFFFFFFFFFFFFFFLL // max "normal integer" on this platform. signed, to allow - this value to reach the minimum signed value
```
Comment (robertoleksy) [inline src/test/xint.cpp]: why not std::numeric_limits?

Comment (rfree-d) [inline src/test/xint.cpp]: 👍 

1) change both defines, to using 
const auto ......... = std numeric_limits
a) max - uint64_t 
b) min - int64_t

2) static assert check that first is equall to this 0xfff..ULL
b) and that min is equall to 8000000000000000 ULL

File: src/test/eint.cpp [hunk 4]
```diff
@@ -28,13 +28,16 @@ TEST(eint, aggrinit_narrow_uchar) {
 	EXPECT_NE(u1,u2);
 }
 
+/*
 TEST(eint, aggrinit_narrow_init) {
 	unsigned int u1=250,u2=251;
-	unsigned int c{ u1-u2 }; // bad: no-compile-error
-	volatile auto unused = c;
+	// ub sanitizer shows error here so it is commented now.
```
Comment (rfree-d) [inline src/test/eint.cpp]: commented out

File: src/test/xint.cpp [hunk 5]
```diff
@@ -438,25 +443,39 @@ void math_tests_overflow_incr(T_INT val) { bool safetype = is_safe_int<T_INT>();
 	_note("Will increment again: " << a );
 	if (safetype) { EXPECT_THROW( func() , std::runtime_error ); }
 	else { // should not fit for given val
-		EXPECT_NO_THROW( func() ); // usafe type fils to throw
-		EXPECT_NE(a , a_corr); // unsafe type has mathematically-invalid value
+		// test nothing for unsafe types, as the test would cause a real UB to happen
+		// and an UB could in theory break all tests and do anything.
+
+		// EXPECT_NO_THROW( func() ); // usafe type fils to throw
```
Comment (rfree-d) [inline src/test/xint.cpp]: my typo  -  fails

File: src/test/eint.cpp (hunk 4 above)
Comment (mikurys) [inline src/test/eint.cpp]: 🆗

File: src/test/xint.cpp (hunk 5 above)
Comment (mikurys) [inline src/test/xint.cpp]: corrected

File: src/test/xint.cpp (hunk 2 above)
Comment (mikurys) [inline src/test/xint.cpp]: 👍

File: src/test/xint.cpp (hunk 1 above)
Comment (mikurys) [inline src/test/xint.cpp]: 👍

File: src/test/xint.cpp (hunk 3 above)
Comment (mikurys) [inline src/test/xint.cpp]: 🆗

**Pull request merged**